cli/src/main/java/com/devonfw/tools/ide/common/JsonVersionItem.java

```diff
@@ -0,0 +1,8 @@
+package com.devonfw.tools.ide.common;
+
+/**
+ * Interface for a data object that can be read from and written to JSON.
+ */
+public interface JsonVersionItem {
+
```

Comment:
I would suggest that we even make it simpler by adding `getVersion()` method here:

```suggestion
  String getVersion();
```

If the version should have a different property name in JSON, you can use Jackson annotation.
Else if the version is transient and needs to be computed out of other attributes, we simply implement it as a getter and annotate with `@JsonIgnore` if needed.
If we later still have a totally unexpected and odd case, we can sill override `String getVersion(JVI jsonVersionItem)` and implement some nonsense for `getVersion()` like `throw new UnsupportedOperationException();` but I guess this will never happen and then we can even prevent to override this method for every JSON updater making it even more simple. WDYT?

Reply:
`getVersion` of the `JsonUrlUpdater` now simply delegates to the `jsonVersionItem.getVersion()` method, and have moved the `getVersion` logic of the `GithubUrlUpdater` and `JavaUrlUpdater` into the appropriate version items.